### PR TITLE
chore: Automatically update acir artifacts

### LINKED ISSUE #2453: Create github action that automatically updates the acir artifacts
### Problem

Currently we sometimes forget to update the artifacts or we update the artifacts and it becomes noise in the PR.

### Happy Case

Add a Github action that automatically creates a PR that updates the artifacts, similar to how release-please updates on each merge to master 

### Alternatives Considered

_No response_

### Additional Context

_No response_

### Would you like to submit a PR for this Issue?

No

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

This is not really tenable at the moment while we are still installing barretenberg.

## Problem\*

resolves #2453 

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
This will fail since we are not installing barretenberg in the action/using nix

**Comment by kevaundray** [comment]
We are no longer installing barretenberg so this is now tenable

**Review by jfecher — COMMENTED** [review]
LGTM but I'll defer to @TomAFrench's review

File: .github/workflows/auto-pr-rebuild-script [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        run: |
+          noirup -p .
+
+      - name: Set up Git user (Github Action)
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+
+      - name: Run rebuild script
+        working-directory: crates/nargo_cli/tests
```
Comment (kevaundray) [inline .github/workflows/auto-pr-rebuild-script:37]:
```suggestion
        working-directory: tooling/nargo_cli/tests
```

File: .github/workflows/auto-pr-rebuild-script [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        run: |
+          chmod +x ./rebuild.sh
+          ./rebuild.sh
+
+      - name: Create or Update PR
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: "Update Acir artifacts"
+          title: "Auto PR - Update ACIR artifacts"
```
Comment (TomAFrench) [inline .github/workflows/auto-pr-rebuild-script:47]:
```suggestion
          title: "chore: Update ACIR artifacts"
```

File: .github/workflows/auto-pr-rebuild-script [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          body: "Automatic PR to update acir artifacts"
+          labels: "auto-pr"
+          branch: "auto-pr-rebuild-script-branch"
```
Comment (TomAFrench) [inline .github/workflows/auto-pr-rebuild-script:50]:
```suggestion
          commit-message: "chore: update acir artifacts"
          title: "chore: Update ACIR artifacts"
          body: "Automatic PR to update acir artifacts"
          labels: "auto-pr"
          branch: "auto-pr-rebuild-script-branch"
```

**Review by TomAFrench — APPROVED** [review]

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**